src/thunderbots/shared/primitive_type.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+/**
+ * This enum represents the IDs of the primitives implemented in firmware.
+ */
+enum class PrimitiveType
```

Comment:
Could we rename to `FirmwarePrimitiveType` just to make it crystal-clear that this is firmware specific?